### PR TITLE
Add PR testing to DigitalOcean

### PR DESCRIPTION
**Pull request by dLobatog**
I think this is all that's needed - can't set the hooks in the repo myself because I don't have permissions, would be glad to do it if I can get them

### CODE REVIEW

File: puppet/modules/slave/templates/test_pull_requests_foreman_digitalocean.json.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "settings": {
+        "test_settings": {
+            "name": "test",
+            "test_prefix": "Test Results:",
+            "allow_multiple": true,
+            "run_without_trigger": true,
+            "branches": {
+                "*": {
+                    "jenkins_job_name": "test_plugin_pull_request",
+                    "downstream_job_name": "test_plugin_foreman-digitalocean_master",
```
Comment (domcleal) [inline puppet/modules/slave/templates/test_pull_requests_foreman_digitalocean.json.erb]: This is foreman_digitalocean (underscore).

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-digitalocean.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+- project:
+    name: foreman-digitalocean
```
Comment (domcleal) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-digitalocean.yaml]: This already exists with an underscore, leave it out.

File: puppet/modules/slave/templates/test_pull_requests_foreman_digitalocean.json.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    "build_token": "<%= jenkins_build_token -%>",
+                    "addtl_jenkins_params": {
+                        "foreman-digitalocean": {
+                            "plugin_repo": "https://github.com/theforeman/foreman-digitalocean.git",
+                            "plugin_name": "foreman-digitalocean"
```
Comment (domcleal) [inline puppet/modules/slave/templates/test_pull_requests_foreman_digitalocean.json.erb]: I think this needs to be underscore, it's the gem name that will be added to the Gemfile to load it.

**Comment by dLobatog** [comment]
Updated to fix the underscores & removed the JJB yaml

**Comment by domcleal** [comment]
Thanks @dLobatog!

**Pull request merged**